Cubical/Homotopy/HSpace.agda

```diff
@@ -27,6 +34,89 @@ record AssocHSpace {ℓ : Level} {A : Pointed ℓ} (e : HSpace A) : Type ℓ whe
                (μ-assoc (pt A) y z)
                refl
 
+record LeftInvHSpace {ℓ : Level} {A : Pointed ℓ} (e : HSpace A) : Type ℓ where
+  constructor LeftInvHSp
+  field
+    μ-equiv : (x : typ A) → isEquiv (HSpace.μ e x)
+
+  open HSpace e
+
+  μ≃ : typ A → typ A ≃ typ A
+  μ≃ x = (μ x , μ-equiv x)
+
+  μ⁻¹ : typ A → typ A → typ A
+  μ⁻¹ x = invIsEq (μ-equiv x)
+
+  μ⁻¹ₗ : (x : typ A) → μ⁻¹ (pt A) x ≡ x
+  μ⁻¹ₗ x = sym (invEq (equivAdjointEquiv (μ≃ (pt A))) (μₗ x))
+
+  μ⁻¹ᵣ : (x : typ A) → μ⁻¹ x x ≡ (pt A)
+  μ⁻¹ᵣ x = sym (invEq (equivAdjointEquiv (μ≃ x)) (μᵣ x))
+
+  -- every left-invertible H-space is strongly homogeneous
+  isHomogeneousHSp : isHomogeneous A
+  isHomogeneousHSp x = pointed-sip A (fst A , x)
+                       ((HSpace.μ e x , μ-equiv x) , HSpace.μᵣ e x)
+
+module _ {ℓ ℓ' : Level} {A : Pointed ℓ} {B : Pointed ℓ'} (hb : HSpace B)
+         (hi : LeftInvHSpace hb) where
+
+  open HSpace hb
+  open LeftInvHSpace hi
+
+  -- this could be given a direct proof
+  →∙HSpace≡ : {f g : A →∙ B} → f .fst ≡ g .fst → f ≡ g
+  →∙HSpace≡ {f} {g} p = →∙Homogeneous≡ isHomogeneousHSp p
+
+{- H-space structures on A are exactly pointed sections of the evaluation map,
+   expressed here with a pointed Π-type -}
+HSpace-Π∙-Iso : {ℓ : Level} (A : Pointed ℓ) → Iso (HSpace A)
+                (Π∙ A (λ x → A →∙ (typ A , x)) (idfun∙ A))
+fst (fst (Iso.fun (HSpace-Π∙-Iso A) e) x) = HSpace.μ e x
+snd (fst (Iso.fun (HSpace-Π∙-Iso A) e) x) = HSpace.μᵣ e x
+fst (snd (Iso.fun (HSpace-Π∙-Iso A) e) i) x = HSpace.μₗ e x i
+snd (snd (Iso.fun (HSpace-Π∙-Iso A) e) i) j = invEq slideSquareEquiv
+  (flipSquare (HSpace.μₗᵣ e)) i j
+HSpace.μ (Iso.inv (HSpace-Π∙-Iso A) s) x = fst (fst s x)
+HSpace.μₗ (Iso.inv (HSpace-Π∙-Iso A) s) x i = fst (snd s i) x
+HSpace.μᵣ (Iso.inv (HSpace-Π∙-Iso A) s) x = snd (fst s x)
+HSpace.μₗᵣ (Iso.inv (HSpace-Π∙-Iso A) s) = flipSquare
+  (equivFun slideSquareEquiv (λ i j → snd (snd s i) j))
+fst (fst (Iso.rightInv (HSpace-Π∙-Iso A) s k) x) = fst (fst s x)
+snd (fst (Iso.rightInv (HSpace-Π∙-Iso A) s k) x) = snd (fst s x)
+fst (snd (Iso.rightInv (HSpace-Π∙-Iso A) s k) i) x = fst (snd s i) x
+snd (snd (Iso.rightInv (HSpace-Π∙-Iso A) s k) i) j = retEq slideSquareEquiv
+  (λ i' j' → snd (snd s i') j') k i j
+HSpace.μ (Iso.leftInv (HSpace-Π∙-Iso A) e k) x = HSpace.μ e x
+HSpace.μₗ (Iso.leftInv (HSpace-Π∙-Iso A) e k) x = HSpace.μₗ e x
+HSpace.μᵣ (Iso.leftInv (HSpace-Π∙-Iso A) e k) x = HSpace.μᵣ e x
+HSpace.μₗᵣ (Iso.leftInv (HSpace-Π∙-Iso A) e k) = flipSquare
+  (secEq slideSquareEquiv (flipSquare (HSpace.μₗᵣ e)) k)
+
+HSpace-Π∙-Equiv : {ℓ : Level} (A : Pointed ℓ) → HSpace A ≃
+                  Π∙ A (λ x → A →∙ (typ A , x)) (idfun∙ A)
+HSpace-Π∙-Equiv A = isoToEquiv (HSpace-Π∙-Iso A)
+
+{- every strongly homogeneous structure on A gives rise
```

Comment:
I'm a bit confused about the terminology: Is ```isHomogenous``` the same as "strongly homogenous"? Maybe a comment at the definition of ```isHomogenous``` could clarify that.